letsencrypt/cli.py

```diff
@@ -1620,6 +1620,11 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):
         "security", "--strict-permissions", action="store_true",
         help="Require that all configuration files are owned by the current "
              "user; only needed if your config is somewhere unsafe like /tmp/")
+    helpful.add(
+        "automation", "--allow-subset-of-names", dest="allow_subset_of_names",
+        action="store_true", default=False,
+        help="Allow subsets of domain names in a single lineage to fail "
+             "validation without exiting.")
```

Comment:
To me, this help isn't very clear. I think we should err of writing too much help rather than two little. I'm not sold on this, but what about something like:
> When performing domain validation, do not consider it a failure if authorizations can not be obtained for a strict subset of the requested domains. This option cannot be used with --csr.

With this in mind, perhaps we should enforce that `--csr` and `--allow-subset-of-domains` aren't used together. Due to the arguments being in different groups, we can't use [argparse's mutual exclusion](https://docs.python.org/2.7/library/argparse.html#mutual-exclusion), but perhaps we should check for this early somewhere in `cli.py` and error out with a useful error message explaining the problem.